db/Pos/DB/Class.hs

```diff
@@ -34,40 +60,40 @@ import           Pos.Core                       (BlockVersionData (..))
 import           Pos.DB.Types                   (DB (..), NodeDBs, blockIndexDB, gStateDB,
                                                  lrcDB, miscDB)
 
-type MonadDB m = (Ether.MonadReader' NodeDBs m, MonadIO m, MonadCatch m)
-
-getNodeDBs :: MonadDB m => m NodeDBs
-getNodeDBs = Ether.ask'
-
-usingReadOptions
-    :: MonadDB m
-    => Rocks.ReadOptions
-    -> ASetter' NodeDBs DB
-    -> m a
-    -> m a
-usingReadOptions opts l =
-    Ether.local' (over l (\db -> db {rocksReadOpts = opts}))
-
-usingWriteOptions
-    :: MonadDB m
-    => Rocks.WriteOptions
-    -> ASetter' NodeDBs DB
-    -> m a
-    -> m a
-usingWriteOptions opts l =
-    Ether.local' (over l (\db -> db {rocksWriteOpts = opts}))
-
-getBlockIndexDB :: MonadDB m => m DB
-getBlockIndexDB = view blockIndexDB <$> getNodeDBs
-
-getUtxoDB :: MonadDB m => m DB
-getUtxoDB = view gStateDB <$> getNodeDBs
+----------------------------------------------------------------------------
+-- Pure
+----------------------------------------------------------------------------
+
+-- | Tag which denotes one of DBs used by application.
+data DBTag
+    = BlockIndexDB
+    | GStateDB
+    | LrcDB
+    | MiscDB
+    deriving (Eq)
+
+dbTagToLens :: DBTag -> Lens' NodeDBs DB
+dbTagToLens BlockIndexDB = blockIndexDB
+dbTagToLens GStateDB     = gStateDB
+dbTagToLens LrcDB        = lrcDB
+dbTagToLens MiscDB       = miscDB
+
+-- | Pure interface to the database.
+-- TODO: add some ways to put something.
+class MonadThrow m => MonadDBPure m where
+    dbGet :: DBTag -> ByteString -> m (Maybe ByteString)
+
+    default dbGet :: (MonadTrans t, MonadDBPure n, t n ~ m) =>
```

Comment:
lol, because it's more reasonable/unified use one approach over all code: or default method or new approach introduced by @int-index.
the current code mix two these approaches